webserver.py

```diff
@@ -116,6 +125,33 @@ def determine_color(uk_time):
         return STATUS_OUT
 
 
+def get_train_details():
+    if not TRANSPORT_APP_KEY or not TRANSPORT_APP_ID or not FAVOURITE_STATION:
+        return {"train_uid": ''}, ''
+    address = "http://transportapi.com/v3/uk/train/station/{}/live.json?" \
+              "app_id={}&app_key={}&darwin=false&train_status=passenger".format(
+        FAVOURITE_STATION,
+        TRANSPORT_APP_ID,
+        TRANSPORT_APP_KEY)
+    print (address)
+
+    # If there's no global departure time set it to now
+    if not DEPARTURE_TIME:
+        departure_time = datetime.now().strftime("%H:%M")
+    else:
+        departure_time = DEPARTURE_TIME
+
+    try:
+        with urllib.request.urlopen(address) as url:
+            data = json.loads(url.read().decode())
+            for train in data["departures"]["all"]:
+                if train["aimed_departure_time"] == departure_time:
+                    return train, data["station_name"]
+            return {"train_uid": ''}, ''
+    except (ValueError, JSONDecodeError):
```

Comment:
if the API is down, blocking requests, or if you don't have internet, the page will return a 500 error. I'd suggest also catching a `urllib.request.URLError` here, then it will still work (Maybe we don't want this to be a silent fail though?)